dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java

```diff
@@ -603,14 +601,19 @@ protected void rawDelete(Context context, Item item) throws AuthorizeException,
         // Remove bundles
         removeAllBundles(context, item);
 
+        // remove version attached to the item
+        removeVersion(context, item);
+
+        //Only clear collections after we have removed everything else from the item
+        item.getCollections().clear();
```

Comment:
I don't undestand how this works.  It just removes content from a List that we're about to throw away with its containing Item.  If this is meant to remove Collection2Item rows, then don't we need to loop over item.getCollections() and CollectionService.removeItem() each?  (But that will delete the Item when the last row is removed....)